### PR TITLE
deprecate skip_children option

### LINKED ISSUE #3893: deprecate skip_children option

### PR DESCRIPTION
**Pull request by scampi**
Close #3893

### CODE REVIEW

File: src/config/config_type.rs [hunk 1]
```diff
@@ -216,6 +227,7 @@ macro_rules! create_config {
                 match key {
                     $(
                         stringify!($i) => {
+                dbg!(stringify!($i));
```
Comment (topecongiro) [inline src/config/config_type.rs]: Could you remove this `dbg!`?

File: src/bin/main.rs [hunk 2]
```diff
@@ -513,6 +515,11 @@ struct GetOptsOptions {
     print_misformatted_file_names: bool,
 }
 
+fn deprecate_skip_children() {
+    let msg = "Option --skip-children is deprecated since it is now the default to not format submodules of given files (#3587)";
```
Comment (topecongiro) [inline src/bin/main.rs]: Could you wrap this string literal to fix within the max-width?

File: src/config/config_type.rs [hunk 3]
```diff
@@ -137,8 +140,16 @@ macro_rules! create_config {
             }
 
             fn fill_from_parsed_config(mut self, parsed: PartialConfig, dir: &Path) -> Config {
+                let deprecate_skip_children = || {
+                    let msg = "Option skip_children is deprecated since it is now the default to not format submodules of given files (#3587)";
```
Comment (topecongiro) [inline src/config/config_type.rs]: And this one as well?

File: src/config/config_type.rs (hunk 1 above)
Comment (scampi) [reply src/config/config_type.rs]: oops

**Comment by scampi** [comment]
@topecongiro the build now correctly fails when test fails https://github.com/rust-lang/rustfmt/pull/3895/commits/d57f94a6ce1102792c6e3153388fd00722b3bf5b

**Comment by scampi** [comment]
The beta build failed because of some reason unrelated to this PR. I'll try to check why.

```
error: invalid suffix `x` for integer literal
 --> tests/source/issue-3779/ice.rs:2:5
  |
2 |     1x;
  |     ^^ invalid suffix `x`
  |
  = help: the suffix must be one of the integral types (`u32`, `isize`, etc)
```

**Comment by calebcartwright** [comment]
That was added as part of https://github.com/rust-lang/rustfmt/pull/3782, though that file is supposed to be ignored

**Comment by calebcartwright** [comment]
There's warning about unstable options in the output just above the failure

```bash
Warning: can't set `unstable_features = true`, unstable features are only available in nightly channel.
Warning: can't set `ignore = IgnoreList { path_set: {"tests/**/issue-3779/ice.rs"}, rustfmt_toml_path: "" }`, unstable features are only available in nightly channel.
```

Any reason why the unstable options wouldn't be working on that one?

**Comment by scampi** [comment]
@calebcartwright I'll have a look about why `ice.rs` is not ignored.

**Comment by calebcartwright** [comment]
@scampi - I believe I found the issue and have opened https://github.com/rust-lang/rustfmt/pull/3897 with fixes. I added these tests but missed the unstable/nightly check which is why they were failing on beta. 

My apologies!

**Pull request merged**